bundles/org.openhab.binding.pulseaudio/src/main/java/org/openhab/binding/pulseaudio/internal/PulseaudioSimpleProtocolStream.java

```diff
@@ -32,86 +40,198 @@
  *
  * @author Gwendal Roulleau - Initial contribution
  * @author Miguel Álvarez - Refactor some code from PulseAudioAudioSink here
+ * @author Miguel Álvarez - Use socket per stream
  *
  */
 @NonNullByDefault
 public abstract class PulseaudioSimpleProtocolStream {
 
     private final Logger logger = LoggerFactory.getLogger(PulseaudioSimpleProtocolStream.class);
 
-    protected PulseaudioHandler pulseaudioHandler;
-    protected ScheduledExecutorService scheduler;
-
-    protected @Nullable Socket clientSocket;
-
-    private ReentrantLock countClientLock = new ReentrantLock();
-    private Integer countClient = 0;
-
-    private @Nullable ScheduledFuture<?> scheduledDisconnection;
+    protected final PulseaudioHandler pulseaudioHandler;
+    protected final ScheduledExecutorService scheduler;
+    protected boolean closed = false;
+    /**
+     * Collect sockets by module id.
+     */
+    protected final Map<Integer, Socket> moduleSockets = new HashMap<>();
+    /**
+     * Collect scheduled disconnection by socket
+     */
+    protected final Map<Socket, ScheduledFuture<?>> scheduledDisconnections = new HashMap<>();
+    /**
+     * Collect created unused modules
+     */
+    protected final Set<ModuleCache> idleModules = new HashSet<>();
+    /**
+     * Collect modules in use by some stream
+     */
+    protected final Set<SimpleProtocolTCPModule> activeModules = new HashSet<>();
 
     public PulseaudioSimpleProtocolStream(PulseaudioHandler pulseaudioHandler, ScheduledExecutorService scheduler) {
         this.pulseaudioHandler = pulseaudioHandler;
         this.scheduler = scheduler;
     }
 
+    /**
+     * Load simple protocol instance
+     *
+     * @throws IOException
+     * @throws InterruptedException when interrupted during the loading module wait
+     */
+    public Optional<SimpleProtocolTCPModule> acquireSimpleProtocolModule(AudioFormat audioFormat)
+            throws IOException, InterruptedException {
+        @Nullable
+        SimpleProtocolTCPModule idleModule = null;
+        synchronized (idleModules) {
+            logger.debug("idle modules: {}", idleModules.size());
+            var cachedModule = idleModules.stream() //
+                    .filter(c -> c.audioFormat.equals(audioFormat)) //
+                    .findFirst().orElse(null);
+            if (cachedModule != null) {
+                idleModule = cachedModule.module;
+                logger.debug("Will try to reuse idle module {}", idleModule.getId());
+                idleModules.remove(cachedModule);
+            }
+        }
+        logger.debug("loading simple protocol tcp module");
+        var spModule = pulseaudioHandler.loadSimpleProtocolModule(audioFormat, idleModule);
+        spModule.ifPresent(activeModules::add);
+        return spModule;
+    }
+
     /**
      * Connect to pulseaudio with the simple protocol
      * Will schedule an attempt for disconnection after timeout
```

Comment:
Is is still true ?
From what I read, disconnection won't happen, unless some sound end playing.

Reply:
I've made some refactor in this method to return a container class that contains the module and runnable instance that allows to release it. I think it makes the code in the consumers easy to write/read. 

After the changes in the processAndComplete sink method, I think the module is correctly released after usage. Thanks for notice it.